tools/bapdoc.ml

```diff
@@ -0,0 +1,232 @@
+    "regular", "Regular.Std", "Regular Data Library";
+    "graphlib", "Graphlib.Std", "Algorithms on graphs";
+    "bitvec", "Bitvec", "Bitvectors and modular arithmetic";
+    "bap-future", "Bap_future.Std", "Futures and Streams";
+    "ogre", "Ogre", "A sexp-based NoSQL database";
+  ];
+
+  "Hardware Specific Libraries", [
+    "bap-arm", "ARM", "ARM-specific definitions";
+    "bap-x86-cpu", "X86_cpu", "x86/x86-64 specific definitions";
+  ];
+
+  "Language and API/ABI Specific Libraries", [
+    "bap-abi", "Bap_abi", "Interface for specifying ABI";
+    "bap-api", "Bap_api", "Interface for defining API";
+    "bap-c", "Bap_c.Std", "Basic definitions of the C language";
+  ];
+
+  "Utility Libraries", [
+    "bap-bml", "Bap_bml", "writing term transformations";
+    "bare", "Bare", "writing rules for matching for Primus observations";
+    "bap-bundle", "Bap_bundle.Std", "creating and opening bundles";
+    "bap-byteweight", "Bap_byteweight", "interface to the Byteweight subsystem";
+    "bap-demangle", "Bap_demangle.Std", "writing name demanglers";
+    "bap-dwarf", "Bap_dwarf.Std", "a native DWARF parser";
+    "bap-ida", "Bap_ida.Std","an interface to IDA Pro";
+    "bap-llvm", "Bap_llvm.Std", "an inteface to LLVM disassemblers and loaders";
+    "bap-plugins", "Bap_plugins.Std", "loading plugins";
+    "bap-recipe", "Bap_recipe", "loading recipes (packs of command line arguments)";
+    "bap-strings", "Bap_strings.Std", "various text utilities";
+    "bap-traces", "Bap_traces.Std", "working with execution traces";
+    "text-tags", "Text_tags", "Extension of Format's semantic tags";
+  ];
+]
+
+let frontends = [
+  "bap", "bap main frontend";
+  "bap-byteweight", "create, obtain and evaluate byteweight signatures";
+]
+
+let introduction = {|
+{2 Introduction}
+
+The Carnegie Mellon University Binary Analysis Platform (CMU BAP) is a
+suite of utilities and libraries that enables analysis of programs in
+their machine representation. BAP is written in OCaml, relies on
+dynamically loaded plugins for extensibility, and is widely used for
+security analysis, program verification, and reverse engineering.
+
+The framework consists of a bunch of libraries, plugins and
+frontends. The libraries provide code reusability, plugins facilitate
+extensibility and frontends serve as entry points.
+
+Frontends come with comprehensive manuals,
+that can be accessed by using [--help] command line options, or via
+the [man] command, if the manpath is configured correctly.
+Finally, you can access a man page for a plugin using
+[--<PLUGIN>-help] command line option of a frontend, e.g., [bap --map-terms-help].
+
+|}
+
+let packages =
+  List.concat_map ~f:(fun (_,ps) -> List.map ~f:(fun (p,_,_) -> p) ps) libraries
+
+let mkdir path =
+  if not (Sys.file_exists path) then
+    Unix.mkdir path 0o770
+
+let run cmd =
+  let res = Sys.command cmd in
+  if res <> 0 then
+    failwith ("Command: '" ^ cmd ^ "' failed")
+
+let dedot name =
```

Comment:
If you just want to chop the .Std suffix, then you can use `String.chop_suffix` if you want to keep Bap.Std untouched you can just match with `"Bap.Std"`. This will make your intentions much more clear. Also, this function should be named `dedot`, it should be `drop_std` or `shorten`.  



However, in general I don't want this function at all, it might beat us later when we will add more libraries. Just extend the libraries specification to include the module name (for the link) and the short name, e.g,
```
"monads", "Monads.Std", "Monads', "The Monads library";
```
You can use an empty string `""` as the default designator, that will mean: take the full name as the short name.